Can't compile with TBB 4.1 and GCC 8.1

I'm trying GCC 8.1 and it's not going well: the system has TBB 4.1 in `/usr/`, and this leads to these errors:
```
/usr/local/gcc-8.1.0/bin/c++   -DBOOST_NO_AUTO_PTR -Isource/numerics -Iinclude -I/home/fac/f/bangerth/p/deal.II/1/dealii/include -I/home/fac/f/bangerth/p/deal.II/1/dealii/bundled/boost-1.62.0/include -I/home/fac/f/bangerth/p/deal.II/1/dealii/bundled/umfpack/UMFPACK/Include -I/home/fac/f/bangerth/p/deal.II/1/dealii/bundled/umfpack/AMD/Include -I/home/fac/f/bangerth/p/deal.II/1/dealii/bundled/muparser_v2_2_4/include -isystem /usr/include/openmpi-x86_64 -isystem /raid/bangerth/bin/petsc-3.7.4/x86_64/include -isystem /raid/bangerth/bin/trilinos-12.8.1/include -isystem /raid/bangerth/bin/petsc-3.7.4/include -isystem /raid/bangerth/bin/nanoflann-1.2.3/include -isystem /raid/bangerth/bin/p4est-2.0.2-a991/FAST/include -isystem /raid/bangerth/bin/slepc-3.7.3/x86_64/include -isystem /raid/bangerth/bin/slepc-3.7.3/include -isystem /raid/bangerth/bin/sundials-2.7.0/include -pedantic -fPIC -Wall -Wextra -Wpointer-arith -Wwrite-strings -Wsynth -Wsign-compare -Wswitch -Woverloaded-virtual -Wsuggest-override -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O2 -funroll-loops -funroll-all-loops -fstrict-aliasing -Wno-unused-local-typedefs -MD -MT source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_project_codim.cc.o -MF source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_project_codim.cc.o.d -o source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_project_codim.cc.o -c /home/fac/f/bangerth/p/deal.II/1/dealii/source/numerics/vector_tools_project_codim.cc
In file included from /usr/include/tbb/enumerable_thread_specific.h:32,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/base/thread_local_storage.h:23,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/base/logstream.h:23,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/vector_memory.h:22,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_full_matrix.h:27,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/base/array_view.h:26,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/base/mpi.h:21,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/distributed/tria_base.h:22,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/numerics/vector_tools.templates.h:27,
                 from /home/fac/f/bangerth/p/deal.II/1/dealii/source/numerics/vector_tools_project_codim.cc:17:
/usr/include/tbb/concurrent_vector.h: In member function ‘tbb::concurrent_vector<T, A>::iterator tbb::concurrent_vector<T, A>::grow_by(tbb::concurrent_vector<T, A>::size_type)’:
/usr/include/tbb/concurrent_vector.h:667:38: error: operands to ?: have different types ‘tbb::internal::concurrent_vector_base_v3::size_type’ {aka ‘long unsigned int’} and ‘tbb::atomic<long unsigned int>’
         return iterator(*this, delta ? internal_grow_by( delta, sizeof(T), &initialize_array, NULL ) : my_early_size);
                                ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/tbb/concurrent_vector.h:667:38: note:   and each type can be converted to the other
/usr/include/tbb/concurrent_vector.h: In member function ‘tbb::concurrent_vector<T, A>::iterator tbb::concurrent_vector<T, A>::grow_by(tbb::concurrent_vector<T, A>::size_type, tbb::concurrent_vector<T, A>::const_reference)’:
/usr/include/tbb/concurrent_vector.h:680:38: error: operands to ?: have different types ‘tbb::internal::concurrent_vector_base_v3::size_type’ {aka ‘long unsigned int’} and ‘tbb::atomic<long unsigned int>’
         return iterator(*this, delta ? internal_grow_by( delta, sizeof(T), &initialize_array_by, static_cast<const void*>(&t) ) : my_early_size);
                                ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/tbb/concurrent_vector.h:680:38: note:   and each type can be converted to the other
```

What's the appropriate work-around for this situation?

--- a/cmake/configure/configure_1_threads.cmake
+++ b/cmake/configure/configure_1_threads.cmake
@@ -138,6 +138,25 @@ MACRO(FEATURE_THREADS_FIND_EXTERNAL var)
   IF(TBB_FOUND)
     SET(${var} TRUE)
   ENDIF()
+
+  #
+  # TBB versions before 4.2 are missing some explicit calls to std::atomic::load
+  # in ternary expressions; these cause compilation errors in some compilers
+  # (such as GCC 8.1 and newer). To fix this we simply blacklist all older
+  # versions:
+  #
+  IF(TBB_VERSION VERSION_LESS "4.2")
+    MESSAGE(STATUS
+      "The externally provided TBB library is older than version 4.2.0, which "
+      "cannot be used with deal.II."
+      )
+    SET(THREADS_ADDITIONAL_ERROR_STRING
+      "The externally provided TBB library is older than version\n"
+      "4.2.0, which is the oldest version compatible with deal.II and its\n"
+      "supported compilers."
+      )
+    SET(${var} FALSE)
+  ENDIF()
 ENDMACRO()
 
 

